ns/log.n3

```diff
@@ -0,0 +1,396 @@
+log:forAll a log:SyntaxArc;  #@@@@@   NOT a real property. A pseudo property
+    rdfs:label     "for All";
+    rdfs:comment """A is true for any object in place of B.
+This is NOT a real rdf property in its behavior - a pseudo-property.
+For examples substitution is not permitted on its object.""";
+    rdfs:domain log:Formula;
+    vs:term_status "unstable".
+
+
+#########################################################
+#
+#  The following are used for inference
+#
+
+log:Truth a rdfs:Class;
+    rdfs:comment """Something which is true: believe it as you would believe this.
+
+Understood natively by cwm in that it will execute rules in a formula
+declared a Truth within a formula it is already taking rules from.""".
+
+#log:Falsehood a rdf:Class;
+#   rdfs:comment "Something which is not true. Not the complement of True!";
+#   rdfs:domain log:context.
+#
+
+log:implies a rdf:Property;
+    rdfs:comment """Logical implication. 
+This is the relation between the antecedent (subject) and
+conclusion (object) of a rule.
+The application of a rule to a knowledge-base is as follows.
+For every substitution which, applied to the antecedent,
+gives a formula which is a subset of the knowledge-base,
+then the result of applying that same substitution to the
+conclusion may be added to the knowledge-base.
+
+related: See log:conclusion. 
+(See the CWM manual for command line options to determine how
+rules from different sources are applied to and the results
+added to various formula.)
+""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+log:includes a rdf:Property;
+    rdfs:comment """The subject formula includes the object formula.
+Formula A includes formula B if there exists some substitution
+which when applied to B creates a formula B' such that for
+every statement in B' is also in A, every variable
+universally (or existentially) quantified in B' is quantified in
+the same way in A. 
+
+Variable substitution is applied recursively to nested compound terms such as
+formulae, lists and sets.
+
+(Understood natively by cwm when in in the antecedent of a rule.
+You can use this to peer inside nested formulae.)""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+log:notIncludes a rdf:Property;
+    rdfs:comment """The object formula is NOT a subset of subject.
+True iff log:includes is false.
+The converse of log:includes. (Understood natively by cwm.
+The subject formula may contain variables.
+(In cwm, variables must of course end up getting bound
+before the log:include test can be done, or an infinite result set
+would result)
+Related: See includes""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+
+# The following connect the inference engine to the web
+
+
+log:parsedAsN3 a rdf:Property;
+    rdfs:comment "The subject string, parsed as N3, gives this formula.";
+    rdfs:label "parses to";
+    rdfs:domain log:String;
```

Comment:
How does this relate to `log:N3Document`?

Reply:
The range is `log:Formula`, but it's unclear to me what the relation is between `log:N3Document` and `log:Formula` ..

Reply:
The way I read the comments on `log:N3Document` and `log:semantics`, I would interpret `log:N3Document` as something whose `log:content` is valid Notation3. That would make it a subclass of `doc:Work`, which is not explicit above...

Reply:
Do you mean, its contents are an N3 string that needs to be parsed?Yes, that seems to better correspond to its definition: 

> A document which, which parsed as Notation3 as defined in general by http://www.w3.org/DesignIssues/Notation3.html and this schema ...

But, then it's not clear why it wasn't used as domain to `log:content` or `log:semantics`, for instance.

Reply:
The domain of `log:content` and `log:semantics` is an external document, presumably of type `doc:Work`, although that seems like an archaic term. In the case of `log:content`, the mime-type is expected to be `text/n3` and be valid Notation3 markup. The range of `log:content` is then a `log:N3Document`, as is the domain of `log:parsedAsN3`. The range of both `log:semantics` and `log:parsedAsN3` is a `log:Formula`.

I was confused by the existence of both `log:N3Document` and `log:N3`. Perhaps `log:N3Document` is meant to be a subclass of `doc:Work`? Perhaps the domain of `log:parsedAsN3` and range of `log:n3String` is meant to be `log:N3`.

If `doc:Work` is intended to represent an external document in general, and `log:N3Document` is meant to be an external document with mime-type `text/n3`, then perhaps we need an intermediate class of documents which can be parsed to result in a `log:Formula`, such as JSON-LD and Turtle.

Reply:
I'd rather say that the range of `log:content` is a xs:string, and its domain is `log:N3Document` (i.e., URIs with that type). Else, `log:N3Document` would be a datatype, no? ..

Reply:
Note that `log:N3` is defined as a datatype, so it seems to make sense that `log:parsedAsN3` would have that in its domain, and `log:n3String` in it's range.

I updated the definition of `log:N3Document` to make it a subtype of `doc:Work`, but it would overly narrow `log:content` to use that in its domain, as it could presumably retrieve any remote document, not just N3. Same with `log:semantics`, but somewhat in between `doc:Work` and `log:N3Document`, as it could work with Turtle.